src/screen/mod.rs

```diff
@@ -0,0 +1,27 @@
+mod boot;
+mod credits;
+mod playing;
+mod title;
+
+use bevy::prelude::*;
+
+pub(super) fn plugin(app: &mut App) {
+    app.init_state::<Screen>()
+        .enable_state_scoped_entities::<Screen>();
+
+    app.add_plugins((
+        boot::plugin,
+        title::plugin,
+        credits::plugin,
+        playing::plugin,
+    ));
+}
+
+#[derive(States, Debug, Hash, PartialEq, Eq, Clone, Default)]
+pub enum Screen {
+    #[default]
+    Boot,
+    Title,
+    Credits,
+    Playing,
+}
```

Comment:
Not a big fan of flattening this,
I'd at least separate `Boot` out of here into a parent state, since you very clearly aren't supposed to go back there.
Including at least one hierarchy here would also help to show how this scales.

Reply:
Fair enough, will be a separate PR.